Content.Shared/Nutrition/EntitySystems/SatiationSystem.cs

```diff
@@ -0,0 +1,396 @@
+    {
+        var amount = _random.Next(
+            (int) component.Thresholds[SatiationThreashold.Concerned] + 10,
+            (int) component.Thresholds[SatiationThreashold.Okay]);
+        SetSatiation(component, amount);
+        UpdateCurrentThreshold(component);
+        DoThresholdEffects(uid, component, alertThresholds, alertCategory, false);
+
+        component.CurrentThreshold = GetThreshold(component, component.Current);
+        component.LastThreshold = SatiationThreashold.Okay; // TODO: Potentially change this -> Used Okay because no effects.
+    }
+
+    private void OnMapInit(EntityUid uid, SatiationComponent component, MapInitEvent args)
+    {
+        InitializeSatiation(uid, component.Thirst, ThirstAlertThresholds, ThirstAlertCategory, args);
+        InitializeSatiation(uid, component.Hunger, HungerAlertThresholds, HungerAlertCategory, args);
+        Dirty(uid, component);
+
+        TryComp(uid, out MovementSpeedModifierComponent? moveMod);
+            _movementSpeedModifier.RefreshMovementSpeedModifiers(uid, moveMod);
+    }
+
+    private void OnShutdown(EntityUid uid, SatiationComponent component, ComponentShutdown args)
+    {
+        _alerts.ClearAlertCategory(uid, HungerAlertCategory);
+        _alerts.ClearAlertCategory(uid, ThirstAlertCategory);
+    }
+
+    private void OnRefreshMovespeed(EntityUid uid, SatiationComponent component, RefreshMovementSpeedModifiersEvent args)
+    {
+        if (_jetpack.IsUserFlying(uid))
+            return;
+
+        if (component.Thirst.CurrentThreshold <= SatiationThreashold.Desperate)
+            args.ModifySpeed(component.Thirst.SlowdownModifier, component.Thirst.SlowdownModifier);
+        if (component.Hunger.CurrentThreshold <= SatiationThreashold.Desperate)
+            args.ModifySpeed(component.Hunger.SlowdownModifier, component.Hunger.SlowdownModifier);
+    }
+
+    private void OnRejuvenate(EntityUid uid, SatiationComponent component, RejuvenateEvent args)
+    {
+        SetThirst(uid, component.Thirst.Thresholds[SatiationThreashold.Okay], component);
+        SetHunger(uid, component.Hunger.Thresholds[SatiationThreashold.Okay], component);
+    }
+
+    /// <summary>
+    /// Adds to the current thirst of an entity by the specified value
+    /// </summary>
+    /// <param name="uid"></param>
+    /// <param name="amount"></param>
+    /// <param name="component"></param>
+    public void ModifyThirst(EntityUid uid, float amount, SatiationComponent? component = null)
+    {
+        if (!Resolve(uid, ref component))
+            return;
+        SetThirst(uid, component.Thirst.Current + amount, component);
+    }
+
+    /// <summary>
+    /// Adds to the current hunger of an entity by the specified value
+    /// </summary>
+    /// <param name="uid"></param>
+    /// <param name="amount"></param>
+    /// <param name="component"></param>
+    public void ModifyHunger(EntityUid uid, float amount, SatiationComponent? component = null)
```

Comment:
```suggestion
    public void ModifyHunger(Entity<SatiationComponent?> ent, float amount)
```

Use `Entity<TComp> ent` instead of `EntityUid uid, TComp comp` for all new functions. (Not just this one!)

Reply:
I replaced all occurrences of EntityUid,XXXComponent with Entity<XXXComponent>